core/src/main/java/org/mule/runtime/core/internal/lifecycle/phases/DefaultLifecycleObjectSorter.java

```diff
@@ -75,13 +80,13 @@ protected int doAddObject(String name, Object object, List<Object> bucket) {
    */
   @Override
   public List<Object> getSortedObjects() {
-    List<Object> sorted = new ArrayList<>(objectCount);
+    Set<Object> sorted = newSetFromMap(new LinkedHashMap<>(objectCount));
```

Comment:
what's the point of doing this?

Reply:
It was supposed to be an IdentityHashMap to avoid returning the same object instance in the list mor that once. 
Changing it...